src/bms/parse/prompt.rs

```diff
@@ -40,9 +40,18 @@ pub trait Prompter {
     fn handle_track_duplication(&self, duplication: TrackDuplication) -> DuplicationWorkaround;
     /// Determines a [`DuplicationWorkaround`] for [`ChannelDuplication`].
     fn handle_channel_duplication(&self, duplication: ChannelDuplication) -> DuplicationWorkaround;
-    /// Shows the user a [`ParseWarningWithRange`].
-    fn warn(&self, warning: ParseWarningWithRange) {
-        eprintln!("{warning:?}");
+}
+
+/// Blanket implementation to allow borrowing a prompter without cloning.
+impl<T: Prompter + ?Sized> Prompter for &T {
+    fn handle_def_duplication(&self, duplication: DefDuplication) -> DuplicationWorkaround {
+        (*self).handle_def_duplication(duplication)
+    }
+    fn handle_track_duplication(&self, duplication: TrackDuplication) -> DuplicationWorkaround {
+        (*self).handle_track_duplication(duplication)
+    }
+    fn handle_channel_duplication(&self, duplication: ChannelDuplication) -> DuplicationWorkaround {
+        (*self).handle_channel_duplication(duplication)
```

Comment:
This implementation should be unnecessary. Because the trait `Prompter` requires only the reference of data which implements it.